trade-mapping-to-ipaffs/src/test/java/uk/gov/defra/stw/mapping/toipaffs/chedd/commodities/CheddPackageTypeMapperTest.java

```diff
@@ -0,0 +1,102 @@
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.CONTAINER_NOT_SPECIFIED;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.CRATE;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.DRUM;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.JAR;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.PACKAGE;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.PAIL;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.PALLET;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.PALLET_BOX;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.TANK;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.TOTE;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.TRAY;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.TUBE;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType.VIAL;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import uk.gov.defra.stw.mapping.dto.CodeType;
+import uk.gov.defra.stw.mapping.dto.PackageTypeCodeType;
+import uk.gov.defra.stw.mapping.dto.PhysicalSpsPackage;
+import uk.gov.defra.tracesx.notificationschema.representation.ComplementParameterSetKeyDataPair;
+import uk.gov.defra.tracesx.notificationschema.representation.enumeration.PackageType;
+
+class CheddPackageTypeMapperTest {
+
+  private CheddPackageTypeKeyDataMapper cheddPackageTypeKeyDataMapper;
+
+  private static final Map<String, PackageType> packageMap = Map.ofEntries(
+      Map.entry("BG", BAG),
+      Map.entry("BL", BALE),
+      Map.entry("OK", BLOCK),
+      Map.entry("BX", BOX),
+      Map.entry("CA", CAN),
+      Map.entry("CT", CARTON),
+      Map.entry("CS", CASE),
+      Map.entry("CK", CASK),
+      Map.entry("CF", COFFER),
+      Map.entry("CN", CONTAINER_NOT_SPECIFIED),
+      Map.entry("CR", CRATE),
+      Map.entry("DR", DRUM),
+      Map.entry("JR", JAR),
+      Map.entry("PK", PACKAGE),
+      Map.entry("PL", PAIL),
+      Map.entry("PX", PALLET),
+      Map.entry("PB", PALLET_BOX),
+      Map.entry("TK", TANK),
+      Map.entry("TT", TOTE),
+      Map.entry("DT", TRAY),
+      Map.entry("TU", TUBE),
+      Map.entry("VI", VIAL));
+
+
+  @BeforeEach
+  void setup() {
+    cheddPackageTypeKeyDataMapper = new CheddPackageTypeKeyDataMapper();
+  }
+
+  @Test
+  void map_ReturnsComplementParameterSetKeyDataPair_WhenComplete() {
+    for (String item : packageMap.keySet()) {
```

Comment:
When you need to test multiple cases it's best to use a ParameterizedTest instead of a for loop inside the test - when this fails you have to work out which case didn't pass instead of just seeing it in the test cases. See `AnimalsCertifiedAsMapperTest` as an example:

```java
@ParameterizedTest
@CsvSource({
    "APPROVED_BODIES, APPROVED",
    "BREEDING, BREEDING",
    "CIRCUS, CIRCUS",
    "FATTENING, FATTENING",
    "OTHER, OTHER",
    "PETS, PETS",
    "QUARANTINE, QUARANTINE",
    "REGISTERED_EQUIDAE, REGISTERED",
    "RELAYING, RELAYING",
    "SLAUGHTER, SLAUGHTER"
})
void map_ReturnsCorrectAnimalCertification(String value, AnimalCertification expected) {
  SpsCertificate spsCertificate = createTestData(value);

  AnimalCertification actual = mapper.map(spsCertificate);

  assertThat(actual).isEqualTo(expected);
}
```

also, try whenever possible to split out the 'given', 'when', 'then' steps of the test - do the actual on one line then assert the actual is expected on another. This makes it clearer what is what is test steps and what is assert steps.